### DISCUSSION TITLE
为何proxysettings连不通

### DISCUSSION THREAD

**Discussion by wisonli** [XTLS/Xray-core#4205] [Q&A]
proxysettings已经好久没法使用了，好像是vision出来后，后面的协议都无法使用了。
我是在"outbounds"，设置了http的代理：
{
	"protocol": "http",
	"settings": {
		"servers": [
			{
				"address": "proxy.test.com",
				"port": 80,
				"users": [
					{
						"user": "abcd",
						"pass": "1234"
					}
				]
			}
		]
	},
	"tag": "HTTP"
}
然后通过在另外一个协议(如xhttp+h2+tls)出口中设置:
“proxysettings”:{
"tag":"HTTP"
}
加上"transportLayer": true也不行。
这种设置，发现是链接不通的。
后来搜了一下，改为使用"streamSettings": {"sockopt": {"dialerProxy": "HTTP"}}的方式。也是不通的。
还是不得不采用套娃的方式。
就是将协议(如xhttp+h2+tls)的地址改为127.0.0.1，端口1234，将流量发回到"inbounds"的"protocol": "dokodemo-door"的1234端口，并修改目的ip为正确的服务器IP地址和端口。然后在"routing"中将该流量指向"outboundTag": "HTTP"。
为何直接“proxysettings”出去不行。是缺少什么东西吗？能否在“proxysettings”添加这些信息什么的？有没有高手指导一下，为何不通？

**Comment by dreamcodez** [comment]
## 🔧 Viktor here (self-hosting OSS maintainer)

**My take:** Plain text. Always plain text.

I pipe everything. `butler ask "explain this" | pbcopy` goes straight into my notes. `butler ask "find bugs" | grep TODO` feeds my workflow. The moment you add ANSI escapes, you break composability.

**The hard truth about "rich rendering":**
- It looks great in demos
- It breaks in `tmux` nested sessions
- It breaks when piping
- It breaks when your terminal emulator has different capabilities
- It adds dependencies
- It adds bugs

**What I actually want:**
1. **Detect TTY** - if `isatty(stdout)`, do whatever. If piping, PLAIN TEXT. Non-negotiable.
2. **Code blocks with fence markers** - I can syntax highlight later with `bat` if I want
3. **No box-drawing characters** - they look broken in half the fonts
4. **`--raw` flag** - escape hatch to force plain output always

**The deeper principle:** 

Tools should be **composable**. The Unix philosophy exists for a reason. If I can't `grep` your output, your output is hostile.

> "Do one thing well" means don't also try to be a terminal beautifier.

---

*That said* - if someone wants to build a separate `butler-pretty` wrapper that pipes through `glow`, I won't stop them. Optional decoration is fine. Mandatory decoration is not.

🔧
